ReactiveArchitecture/Classes/nowplaying/viewcontroller/NowPlayingViewController.swift

```diff
@@ -176,61 +180,61 @@ class NowPlayingViewController: UIViewController {
         //
         pageNumber = uiModel.pageNumber
         
+        
         //
         //Update adapter
         //
-        if (self.nowPlayingTableViewController!.tableView.isHidden) {
+        if self.nowPlayingTableViewController.tableView.isHidden {
             
             //Process last adapter command
-            if (uiModel.adapterCommandType == AdapterCommandType.ADD_DATA) {
-                nowPlayingTableViewController!.addAll(listToAdd: uiModel.resultList!)
-            } else if (uiModel.adapterCommandType == AdapterCommandType.SHOW_IN_PROGRESS) {
-                nowPlayingTableViewController?.add(itemToAdd: nil)
+            if uiModel.adapterCommandType == .addData, let list = uiModel.resultList {
+                nowPlayingTableViewController.addAll(listToAdd: list)
             }
             
             //make table visible
-            self.nowPlayingTableViewController!.tableView.isHidden = false
+            self.nowPlayingTableViewController.tableView.isHidden = false
             
             //Trigger first load
-            let scrollEvent: ScrollEvent = ScrollEvent.init(pageNumber: self.pageNumber! + 1)
+            let scrollEvent: ScrollEvent = ScrollEvent.init(pageNumber: self.pageNumber + 1)
             nowPlayingViewModel?.processUiEvent(uiEvent: scrollEvent)
         } else {
-            if (uiModel.adapterCommandType == AdapterCommandType.ADD_DATA) {
+            if uiModel.adapterCommandType == .addData {
```

Comment:
Didn't clean this up everywhere but I generally prefer removing the prefix where it's implicit

Reply:
so if I start typing "." on Xcode it'll expand all the enum types? I guess I'll find out...
:thumbsup: